src/Atoms/Switch/Switch.jsx

```diff
@@ -0,0 +1,264 @@
+        padding:      '4px 11px',
+        marginBottom: 0,
+        textAlign:    'center',
+        whiteSpace:   'nowrap',
+        cursor:       'pointer',
+        userSelect:   'none',
+        position:     'absolute',
+        margin:       0,
+        top:          0,
+      },
+      '&:not(.switch-checked) .switch-group': {
+        left: '-100%',
+      },
+      '&:not(.disabled):hover': {
+        borderColor: '#a4a4a4',
+      },
+      '& > .switch-group': {
+        textAlign:        'center',
+        whiteSpace:       'nowrap',
+        cursor:           'pointer',
+        position:         'absolute',
+        width:            '200%',
+        top:              0,
+        bottom:           0,
+        left:             0,
+        transition:       'left 0.35s',
+        '& > .switch-on': {
+          left:         0,
+          paddingRight: '24px',
+          right:        '50%',
+        },
+        '& > .switch-off': {
+          left:            '50%',
+          paddingLeft:     '23px',
+          right:           0,
+          backgroundColor: '#e6e6e6',
+        },
+        '& > .switch-slider': {
+          backgroundColor: '#fff',
+          position:        'relative',
+          margin:          '0 auto',
+          width:           '2px',
+          padding:         '0px 10px',
+          height:          '100%',
+          borderRadius:    '4px',
+          borderWidth:     '0px 1px',
+          borderColor:     '#ccc',
+          borderStyle:     'solid',
+        },
+      },
+      // Sizes
+      '&.switch-large': {
+        width:               '57px',
+        height:              '31px',
+        '& > .switch-group': {
+          '& .switch-on': {
+            paddingTop:  '6px',
+            paddingLeft: '12px',
+          },
+          '& .switch-off': {
+            paddingTop:  '6px',
+            paddingLeft: '24px',
+          },
+          '& .switch-slider': {
+            paddingTop:    '0px',
+            paddingBottom: '0px',
+            width:         '4px',
+          },
+        },
+      },
+      '&.switch-small': {
+        width:               '34px',
+        height:              '20px',
+        fontSize:            '11px',
+        '& > .switch-group': {
+          '& .switch-on': {
+            paddingTop:  '2px',
+            paddingLeft: '7px',
+          },
+          '& .switch-off': {
+            paddingTop:  '2px',
+            paddingLeft: '12px',
+          },
+          '& .switch-slider': {
+            width:   '1px',
+            padding: '0px 5px',
+          },
+        },
+      },
+      // Disabled
+      '&.disabled': {
+        borderColor: '#dedede',
+        cursor:      'not-allowed',
+        '& *':       {
+          cursor: 'not-allowed',
+        },
+        '& > .switch-group': {
+          '& > .switch-off': {
+            backgroundColor: '#efefef',
+            color:           '#747474',
+          },
+        },
+      },
+      // Color stuff
```

Comment:
this color stuff sounds like big shit, since we're in a callback function now, why don't you factorize it as you previously did ?

something like :

```js
const colors = ['primary', 'success', 'default'].map(v => ({
  `&.switch-color-${v}`: {
    '&.switch-checked': {
      borderColor: variants[v].borderColor,
    },
    ...
  },
}))

return {
  root: {
    [...]
  },
  ...colors,
}
```